src/harness/testcases/heartbeat_testcase.py

```diff
@@ -78,6 +78,198 @@ def test_10_9_4_1_1_1(self):
+    self.assertEqual(response['response']['responseCode'], 0)
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_HBT_4(self):
+    """Request grant renewal from heartbeat request (3 devices).
+
+    Returns response code 0 (NO_ERROR)
+    """
+    # Register the devices
+    registration_request = []
+    for device_filename in ('device_a.json', 'device_b.json', 'device_c.json'):
+      device = json.load(
+          open(os.path.join('testcases', 'testdata', device_filename)))
+      self._sas_admin.InjectFccId({'fccId': device['fccId']})
+      registration_request.append(device)
+    request = {'registrationRequest': registration_request}
+    response = self._sas.Registration(request)['registrationResponse']
+    # Check registration response
+    cbsd_ids = []
+    for resp in response:
+      self.assertEqual(resp['response']['responseCode'], 0)
+      cbsd_ids.append(resp['cbsdId'])
+    del request, response
+
+    # Request grant
+    grant_request = []
+    for grant_filename, cbsd_id in zip(['grant_0.json', 'grant_0.json',
+                                    'grant_0.json'], cbsd_ids):
+      grant = json.load(
+          open(os.path.join('testcases', 'testdata', grant_filename)))
+      grant['cbsdId'] = cbsd_id
+      grant_request.append(grant)
+    request = {'grantRequest': grant_request}
+    # Check grant response
+    response = self._sas.Grant(request)['grantResponse']
+    grant_ids = []
+    grant_expire_times = []
+    for response_num, resp in enumerate(response):
+      self.assertEqual(resp['cbsdId'], cbsd_ids[response_num])
+      self.assertGreater(len(resp['grantId']), 0)
+      self.assertEqual(resp['response']['responseCode'], 0)
+      grant_ids.append(resp['grantId'])
+      grant_expire_times.append(datetime.strptime(resp['grantExpireTime'],
+                                                  '%Y-%m-%dT%H:%M:%SZ'))
+    del request, response
+
+    # Heartbeat requests with grantRenew set to True
+    heartbeat_request = []
+    for cbsd_id, grant_id in zip(cbsd_ids, grant_ids):
+      heartbeat_request.append({
+          'cbsdId': cbsd_id,
+          'grantId': grant_id,
+          'operationState': 'GRANTED',
+          'grantRenew': True
+      })
+    request = { 'heartbeatRequest': heartbeat_request}
+    response = self._sas.Heartbeat(request)['heartbeatResponse']
+    # Check the heartbeat response
+    for response_num, resp in enumerate(response):
+      transmit_expire_time = datetime.strptime(resp['transmitExpireTime'],
+                                               '%Y-%m-%dT%H:%M:%SZ')
+      self.assertEqual(resp['cbsdId'], cbsd_ids[response_num])
+      self.assertEqual(resp['grantId'], grant_ids[response_num])
+      self.assertLess(datetime.utcnow(), transmit_expire_time)
+      self.assertLessEqual((transmit_expire_time -
+                       datetime.utcnow()).total_seconds(), 240)
+      self.assertLess(transmit_expire_time, grant_expire_times[response_num])
```

Comment:
The times could be the same.
self.assertLessEqual(transmit_expire_time, grant_expire_times[response_num])